thunder/core/transforms.py

```diff
@@ -3478,6 +3480,14 @@ def _vjp(primals, cotangents, **kwargs):
         flat_func, flat_args, spec = flatten_func(func, primals, kwargs)
         trace = construct_trace()(flat_func, *flat_args)
         result, vjp_result = vjp_call(flat_args, cotangents, trace=trace)
+        vjp_result = tuple(
```

Comment:
Hi @IvanYashchuk , when calculating `binary_ops(cpu_scalar_tensor, cuda_tensor)`, we want to skip the broadcast for cpu_scalar_tensor, and in the runtime backend(nvfuser/torch) can take the mix of cpu_scalar_tensor and cuda_tensor to do the calculation. The problem is in the backward, the bwd of broadcast will generate a sum for the case like cpu_scalar_tensor, now the broadcast of cpu_scalar_tensor is ignored in forward, this sum doesn't appear in the bwd anymore so the grad of scalar cpu tensor is wrong.
I tried to do the sum here for grad of cpu scalar tensor

Reply:
good catch!

Reply:
Thank you for the explanation! It makes sense and the approach is correct.